Marlin/src/feature/easythreed_ui.cpp

```diff
@@ -188,46 +196,50 @@ void EasythreedUI::printButton() {
             blink_interval_ms = LED_BLINK_2;                        // Blink the indicator LED at 1 second intervals
             print_key_flag = PF_PAUSE;                              // The "Print" button now pauses the print
+            print_flag = true;                                      // Set UI print state flag
             card.mount();                                           // Force SD card to mount - now!
             if (!card.isMounted) {                                  // Failed to mount?
-                blink_interval_ms = LED_OFF;                        // Turn off LED
-                print_key_flag = PF_START;
-                return;                                             // Bail out
+              blink_interval_ms = LED_OFF;                          // Turn off LED
+              print_key_flag = PF_START;
+              print_flag = false;                                   // Set UI print state flag
+              return;                                               // Bail out
+            }
+            card.ls();                                              // List all files to serial output
+            const uint16_t filecnt = card.countFilesInWorkDir();    // Count printable files in cwd
+            if (filecnt) {                                          // Any printable files?
+              card.selectFileByIndex(filecnt);                      // Select the last file according to current sort options
+              card.openAndPrintFile(card.filename);                 // Start printing it
             }
-            card.ls();                                            // List all files to serial output
-            const uint16_t filecnt = card.countFilesInWorkDir();  // Count printable files in cwd
-            if (filecnt == 0) return;                             // None are printable?
-            card.selectFileByIndex(filecnt);                      // Select the last file according to current sort options
-            card.openAndPrintFile(card.filename);                 // Start printing it
             break;
           }
-          case PF_PAUSE: {                                          // Pause printing (not currently firing)
-            if (!printingIsActive()) break;
-            blink_interval_ms = LED_ON;                             // Set indicator to steady ON
-            queue.inject(F("M25"));                                 // Queue Pause
-            print_key_flag = PF_RESUME;                             // The "Print" button now resumes the print
-            break;
+          case PF_PAUSE: {                                          // Pause printing
+            if (IS_SD_PRINTING()) {                                 // If pause is during heatup, heatup will continue and then pause before printing
+              card.pauseSDPrint();                                  // This is more reliable than queueing M25 ??
```

Comment:
Note that this is a hack, as we should always use the Marlin-wide options to pause / resume printing so that the head will park, heaters can time out, etc. Pausing the SD card processing directly is not enough to get the benefit of safety behaviors. If you are experiencing problems with pausing using proper methods, then the deficits of those proper methods need to be addressed instead of resorting to hacks.

Reply:
Understood. 
I don't have much time in the near future for progressing this code, so if it makes sense to you I'll close this PR and raise a new one when I have some updates.